cmd/amp-cluster/bootstrap-docker.go

```diff
@@ -0,0 +1,308 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"github.com/blang/semver"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+	dockerclient "github.com/docker/docker/client"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+var DockerChannels = [...]string{"main"}
+
+//var DockerChannels = [...]string{"main", "testing"}
+
+const (
+	minimumDockerEngineVersion = "1.12.3"
+	dockerAPIMinVersion        = "1.24"
+	dockerUrl                  = "unix:///var/run/docker.sock"
+	dockerGetUrl               = "http://get.docker.com"
+	dockerRemoteAPIPort        = "2375"
+)
+
+var startCommands = [...]string{"systemctl start docker.service", "service docker start", "start docker"}
+var enableCommands = [...]string{"systemctl enable docker.service", "chkconfig docker on", "update-rc.d docker defaults"}
+
+func (c *clusterClient) getDockerServerVersion() (string, error) {
+	// first try the Docker api
+	if c.dockerClient == nil {
+		defaultHeaders := map[string]string{"User-Agent": "amp-bootstrap"}
+		c.dockerClient, _ = dockerclient.NewClient(dockerUrl, dockerAPIMinVersion, nil, defaultHeaders)
+	}
+	if c.dockerClient != nil {
+		version, err := c.dockerClient.ServerVersion(c.ctx)
+		if err == nil {
+			return version.Version, nil
+		}
+	}
+	log.Println("Unable to contact the Docker server through the API: %s, switch to the Docker cli")
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	versionCmd := [...]string{"docker", "version", "-f", "{{ .Server.Version }}"}
+	cmd := exec.Command(versionCmd[0], versionCmd[1:]...)
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
```

Comment:
if you want to see output you can also use:
    cmd.Stdout = os.Stdout
    cmd.Stderr = os.Stderr